### PR TITLE
Resolve callback hell

### PR DESCRIPTION
**Pull request by suryakun**
I have two commit at this PR. first is my codes without jscs config, that was nice indentations.
and the second is my codes with jscs config. a little bit annoying with the indentation, but the jscs's rules force me to use that weird style. I not rebase my commits so it will trackable.

please let me know if this is not satisfied.

### CODE REVIEW

File: parse-promise.js [hunk 1]
```diff
@@ -1 +1,82 @@
...
+const moduleCsv = require('./class-csv');
+const async  = require('async');
+
+class PromiseParse {
+
+	readFile() {
+		return new Promise(resolve  => {
+			fs.readFile(__dirname + '/sample.csv', (err, loadedCsv) => {
+				parse(loadedCsv, (err, parsed) => {
+					if (err) callback(new Error('Error reading file'));
```
Comment (diorahman) [inline parse-promise.js]: `callback`? where did you define this `callback`? And how to handle the error at the caller site?

File: parse-promise.js [hunk 2]
```diff
@@ -1 +1,82 @@
...
+	main() {
+		var self = this;
+		self.readFile()
+			.then((parsed) => {
+				self.loopingLine(parsed, (line) => {
+					self.sendSms(line).then((line) => {
+						self.logToS3(line);
+					});
+				});
+			});
```
Comment (diorahman) [inline parse-promise.js]: Please use promise's facilities in here. We should fight the so called callback hell. However the shown code has similar pyramid of doom :smile:

File: parse-stream.js [hunk 3]
```diff
@@ -1,3 +1,88 @@
...
+	readFile(callback) {
+		let delay = 0;
+		let rl = readline.createInterface({
+			input: fs.createReadStream(__dirname + '/sample.csv')
+		});
+
+		rl.on('line', line => {
+			parse(line, (err, parsed) => {
+				if (delay == 0) {
+					delay=1; // skip first line of stream
```
Comment (diorahman) [inline parse-stream.js]: Why did you choose `delay` as variable name?

Comment (suryakun) [inline parse-stream.js]: Ugh, sorry for my messy code. already fixed now. hope it good enought.

Thank you for your fun challenge, please let me know if some not satisfied again.

**Comment by diorahman** [comment]
Thank you @suryakun we will get back to you after some internal processes.

**Pull request closed without merging**